src/firebase-social-provider.js

```diff
@@ -243,19 +243,17 @@ FirebaseSocialProvider.prototype.addUserProfile_ = function(friend) {
   // about a client when they send us the first message (e.g. instance message
   // in the case of uProxy).
   var clients = new Firebase(this.getClientsUrl_(friend.userId));
-  clients.on('value', function(value) {
-    // TODO: this is going to be updating the lastUpdated and lastSeen values
-    // for each client, any time any of the clients change!
-    // i.e. if there are only 2 clients A and B, then C gets added, this will be
-    // invoked with A, B, and C..  Find a way to only pay attention to C.
-    // possibly using some combination of child_added and child_changed instead
-    // of value.
-    // See: https://www.firebase.com/docs/web/api/query/on.html
-    value.forEach(function(snapshot) {
-      var clientId = friend.userId + '/' + snapshot.key();
-      var status = snapshot.val() == 'ONLINE' ? 'ONLINE' : 'OFFLINE';
-      this.addOrUpdateClient_(friend.userId, clientId, status);
-    }.bind(this));
+  clients.on('child_added', function(snapshot) {
```

Comment:
Was this potentially receiving other events of importance when it was on 'value' and not on 'child_added'?